context/app/static/js/components/detailPage/visualization/VisualizationShareButton/utils.ts

```diff
@@ -12,13 +12,14 @@ const getUrl = (conf: object, onOverMaximumUrlLength: () => void) => {
 };
 
 const copyToClipBoard = (conf: object, onOverMaximumUrlLength: () => void) => {
-  const dummy = document.createElement('input');
-  document.body.appendChild(dummy);
   const url = getUrl(conf, onOverMaximumUrlLength);
-  dummy.setAttribute('value', url);
-  dummy.select();
-  document.execCommand('copy');
-  document.body.removeChild(dummy);
+
+  navigator.clipboard
+    .writeText(url)
+    .then()
+    .catch((err) => {
+      console.error('Failed to copy text to clipboard: ', err);
```

Comment:
This seems like it'd be worth alerting the user about via a toast; maybe worth adding a `onCopyError` prop that runs `toastError` on the error text?

Reply:
Ah, can't we just use the existing `useCopyText` hook?

Reply:
Oh, right - I somehow forgot that exists separately from this 😆 Yeah, the `useHandleCopyClick` hook from `useCopyText` actually does exactly that.

Reply:
Oh fantastic!